server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/UserTokenGranterTest.java

```diff
@@ -188,6 +194,15 @@ protected Authentication validateRequest(TokenRequest request) {
 
     @Test
     public void happy_day() {
+        when(clientAuthenticationProvider.authenticate(any(UsernamePasswordAuthenticationToken.class)))
+            .thenThrow(new BadCredentialsException("Bad credentials"));
+        missing_parameter("non existent");
+    }
+
+    @Test(expected = InvalidClientException.class)
+    public void test_client_no_secret() {
+        when(clientAuthenticationProvider.authenticate(any(UsernamePasswordAuthenticationToken.class)))
+            .thenReturn(null);
         missing_parameter("non existent");
```

Comment:
I know this method was already there, but can we improve the name of missing_parameter()? Maybe validate_with_missing_parameter()?